app/src/main/java/com/keylesspalace/tusky/components/account/AccountViewModel.kt

```diff
@@ -97,37 +99,47 @@ class AccountViewModel @Inject constructor(
 
     fun changeFollowState() {
         val relationship = relationshipData.value?.data
-        if (relationship?.following == true || relationship?.requested == true) {
-            changeRelationship(RelationShipAction.UNFOLLOW)
-        } else {
-            changeRelationship(RelationShipAction.FOLLOW)
+        viewModelScope.launch {
+            if (relationship?.following == true || relationship?.requested == true) {
+                changeRelationship(RelationShipAction.UNFOLLOW)
+            } else {
+                changeRelationship(RelationShipAction.FOLLOW)
+            }
         }
     }
 
     fun changeBlockState() {
-        if (relationshipData.value?.data?.blocking == true) {
-            changeRelationship(RelationShipAction.UNBLOCK)
-        } else {
-            changeRelationship(RelationShipAction.BLOCK)
+        viewModelScope.launch {
```

Comment:
moving the `viewModelScope.launch` into `changeRelationship` would save quite some lines

Reply:
Done.